onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -313,21 +313,26 @@ def initializers(self) -> Mapping[str, torch.Tensor]:
 
     @beartype
     def add_input(
-        self, input_name: str, input_value: Optional[torch.Tensor] = None
+        self,
+        input_name: Optional[str],
+        shape: Optional[Union[torch.Size, Sequence[Union[int, str, None]]]] = None,
+        dtype: Optional[torch.dtype] = None,
     ) -> TorchScriptTensor:
-        # TODO: Take in a TorchScriptTensor?
-        # TODO: Support dynamic shapes
-        if input_value is None:
+        if input_name is None:
             # This input argument is None, which is mapped
             # to a NULL value in TorchScript type system.
             torch_value = _create_op_call_in_torch_graph(
                 self._torch_graph, "prim::Constant", inputs=(), attributes={}
             )[0]
             torch_value.setType(torch.OptionalType.ofTensor())
-            tensor_value = _wrap_torch_value_to_tensor(torch_value)
-            return tensor_value  # type: ignore[return-value]
-        torch_value = self._torch_graph.addInput(input_name)
-        torch_value.setType(torch.TensorType.create_from_tensor(input_value))
+        else:
+            torch_value = self._torch_graph.addInput(input_name)
+            torch_value.setType(torch_value.type().with_dtype(dtype))  # type: ignore[arg-type]
+            torch_value.setType(
+                torch_value.type().with_sizes(
+                    [dim if isinstance(dim, int) else None for dim in shape]  # type: ignore[union-attr]
```

Comment:
Can you add a TODO or NOTE or FIXME that this step loses the information of individual symbols?

Reply:
OK